iceberg/iceberg-handler/src/main/java/org/apache/iceberg/mr/hive/HiveIcebergOutputCommitter.java

```diff
@@ -231,25 +314,47 @@ public void commitJob(JobContext originalContext) throws IOException {
    */
   @Override
   public void abortJob(JobContext originalContext, int status) throws IOException {
-    JobContext jobContext = TezUtil.enrichContextWithVertexId(originalContext);
-    JobConf jobConf = jobContext.getJobConf();
+    abortJobs(Collections.singletonList(originalContext));
+  }
+
+  public void abortJobs(List<JobContext> jobContexts, JobStatus.State runState) throws IOException {
+    int state = runState.getValue();
+    if (state != JobStatus.FAILED && state != JobStatus.KILLED) {
```

Comment:
Why is this invalid?

Reply:
This logic comes from hadoop.
https://github.com/apache/hadoop/blob/a177232ebc3be826774701aefe63ed30a6b36fe7/hadoop-mapreduce-project/hadoop-mapreduce-client/hadoop-mapreduce-client-core/src/main/java/org/apache/hadoop/mapred/OutputCommitter.java#L304

I think only a running job can be aborted.

Reply:
Isn't it only us who is calling this method, and we only call it with `FAILED`.

Reply:
True, I added this validation just to be consistent with the hadoop implementation. I don't think it is harmful.